packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -339,14 +347,28 @@ const SynapseGrid = forwardRef<SynapseGridHandle, SynapseGridProps>(
                           'row-selected': selectedRowIndex === rowIndex,
                         })
                       }
-                      cellClassName={({ rowData, rowIndex, columnId }) =>
-                        getCellClassName({
+                      cellClassName={({ rowData, rowIndex, columnId }) => {
+                        const baseCellClassName = getCellClassName({
                           rowData: rowData as DataGridRow,
                           rowIndex,
                           columnId,
                           selectedRowIndex,
                         })
-                      }
+
+                        // Add selection styling based on lastSelection
+                        const isInSelection =
+                          lastSelection &&
+                          rowIndex >= lastSelection.min.row &&
+                          rowIndex <= lastSelection.max.row &&
+                          colValues.findIndex(col => col.id === columnId) >=
+                            lastSelection.min.col &&
+                          colValues.findIndex(col => col.id === columnId) <=
+                            lastSelection.max.col
```

Comment:
Should probably just pass `isInSelection` to `getCellClassName`, and `getCellClassName` would apply the correct classname

Reply:
You read my mind! I just updated `getCellClassName` and its unit tests. Feel free to take a look, otherwise I'll merge when the actions finish.